DataFormats/EcalDigi/interface/EcalEBPhase2TriggerPrimitiveDigi.h

```diff
@@ -0,0 +1,65 @@
+#ifndef ECALEBPHASE2TRIGGERPRIMITIVEDIGI_H
+#define ECALEBPHASE2TRIGGERPRIMITIVEDIGI_H 1
+
+#include <ostream>
+#include <vector>
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDigi/interface/EcalEBPhase2TriggerPrimitiveSample.h"
+
+/** \class EcalEBPhase2TriggerPrimitiveDigi
+\author N. Marinelli - Univ. of Notre Dame
+
+
+*/
+
+class EcalEBPhase2TriggerPrimitiveDigi {
+public:
+  typedef EBDetId key_type;  ///< For the sorted collection
+
+  EcalEBPhase2TriggerPrimitiveDigi();  // for persistence
+  EcalEBPhase2TriggerPrimitiveDigi(const EBDetId& id);
+
+  void swap(EcalEBPhase2TriggerPrimitiveDigi& rh) {
+    std::swap(id_, rh.id_);
+    std::swap(size_, rh.size_);
+    std::swap(data_, rh.data_);
+  }
+
+  const EBDetId& id() const { return id_; }
+  int size() const { return size_; }
+
+  const EcalEBPhase2TriggerPrimitiveSample& operator[](int i) const { return data_[i]; }
+  const EcalEBPhase2TriggerPrimitiveSample& sample(int i) const { return data_[i]; }
+
+  void setSize(int size);
+  void setSample(int i, const EcalEBPhase2TriggerPrimitiveSample& sam);
+  void setSampleValue(int i, uint16_t value) { data_[i].setValue(value); }
+
+  static const int MAXSAMPLES = 20;
```

Comment:
It's a leftover from the past. Infact also for the Phase I I think that number is wrong, since for PhaseI TP 5 samples were foreseen ( these are the sample of the TP, not of the incoming digis). I believe that it wastes space but is not relevant to the functioning.

Reply:
I am not much worried about the wasted space but about the clarity of the code. If the number is not up to date anymore please update it.